src/eos-shard-writer.c

```diff
@@ -19,10 +19,19 @@
 
 #include "eos-shard-writer.h"
 
-#include <endian.h>
 #include <fcntl.h>
 #include <string.h>
 
+#include "config.h"
+#ifdef HAVE_ENDIAN_H
+# include <endian.h>
+#elif defined(HAVE_LIBKERN_OSBYTEORDER_H)
+# include <libkern/OSByteOrder.h>
+# define htole64(x) OSSwapHostToLittleInt64(x)
```

Comment:
i'm half-tempted to use something like https://gist.github.com/panzi/6856583

I really like the htole64 family of macros much better than bswap64 and such.